tests/unittests/test_error_handling.py

```diff
@@ -0,0 +1,439 @@
+import unittest
+import tap_twitter_ads
+import tap_twitter_ads.client as client
+from unittest import mock
+from twitter_ads.client import Client
+# from tap_twitter_ads.sync import get_resource, post_resource
+from tap_twitter_ads.streams import TwitterAds
+
+
+
+class Mockresponse(Exception):
+    def __init__(self, status_code, details=[]):
+        self.code = status_code
+        self.details = details
+
+@mock.patch("tap_twitter_ads.streams.Request")
+class TestExceptionHandling(unittest.TestCase):
+    """
+        Test cases to verify error is raised with proper message  for get_resource method.
+    """
+    
+    client_obj = Client(
+        consumer_key='test',
+        consumer_secret='test',
+        access_token='test',
+        access_token_secret='test'
+    )
+    
+    test_stream = TwitterAds()
+
+    @mock.patch("tap_twitter_ads.streams.Cursor", side_effect=Mockresponse(400))
+    def test_400_error_custom_message(self, mocked_cursor, mocked_request):
+        """
+            Test case to verify 400 error message from response
+        """
+        
+        with self.assertRaises(client.TwitterAdsBadRequestError) as e:
+            self.test_stream.get_resource("test", self.client_obj, "path")
+
+        self.assertEqual(str(e.exception), "HTTP-error-code: 400, Message: The request is missing or has a bad parameter.")
+    
+    @mock.patch("tap_twitter_ads.streams.Cursor", side_effect=Mockresponse(400, [{"message":"This mesaage from response 400"}]))
```

Comment:
Spelling mistake `mesaage` in all unittests, please correct.

Reply:
Resolve spell mistakes in unit tests.